controllers/machinedeletion_controller.go

```diff
@@ -63,19 +73,54 @@ func (r *MachineDeletionReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	if remediation = r.getRemediation(ctx, req); remediation == nil {
 		return ctrl.Result{}, nil
 	}
-	//not a machine based remediation
-	machineOwnerRef := getMachineOwnerRef(remediation)
-	if machineOwnerRef == nil {
-		return ctrl.Result{}, nil
+
+	var machine *unstructured.Unstructured
+	//Health check was done by NHC
+	if node, err := r.getNodeFromMdr(remediation); err == nil {
+		if machine, err = r.buildMachineFromNode(node); err != nil {
+			return ctrl.Result{}, err
+		}
+		if isMachineBelongToMasterNode(machine) {
+			return ctrl.Result{}, nil
+		}
+
+	} else { //Failed both in fetching the machine and in fetching the node
+		return ctrl.Result{}, err
 	}
-	//delete the machine
-	if err := r.deleteMachine(ctx, buildMachine(machineOwnerRef, remediation)); err != nil {
+
+	if err := r.deleteMachineOfNode(ctx, machine, remediation.Name); err != nil {
 		return ctrl.Result{}, err
 	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *MachineDeletionReconciler) deleteMachineOfNode(ctx context.Context, machine *unstructured.Unstructured, nodeName string) error {
+	//delete the machine
+	_ = r.Client.Delete(ctx, machine)
```

Comment:
why error is ignored?
if there's an err I'd expect a log message

Reply:
The error is ignored because I verify the deletion in the next paragraph so I think it's redundant.
I think you are right about the log  (but IMO it should be in the next paragraph).

Reply:
the err here will be able to explain why deletion failed, not only that it failed.

Reply:
Sure, fair enough :+1: